packages/next/server-actions/index.ts

```diff
@@ -0,0 +1,80 @@
+'use server';
+
+import { type GetContextParameters, type InteractionMode } from '@logto/node';
+import NodeClient from '@logto/node/edge';
+
+import BaseClient from '../src/client';
+import { createSession } from '../src/session';
+import type { LogtoNextConfig } from '../src/types.js';
+
+export type { LogtoContext, InteractionMode } from '@logto/node';
+
+export default class LogtoClient extends BaseClient {
+  constructor(config: LogtoNextConfig) {
+    super(config, {
+      NodeClient,
+    });
+  }
+
+  handleSignIn = async (
+    cookie: string,
+    redirectUri: string,
+    interactionMode?: InteractionMode
+  ): Promise<{ url: string; newCookie?: string }> => {
+    const { nodeClient, session } = await this.createNodeClientFromHeaders(cookie);
+    await nodeClient.signIn(redirectUri, interactionMode);
+
+    if (!this.navigateUrl) {
+      // Not expected to happen
+      throw new Error('navigateUrl is not set');
+    }
+
+    return {
+      url: this.navigateUrl,
+      newCookie: await session.getValues?.(),
+    };
+  };
+
+  handleSignOut = async (cookie: string, redirectUri = this.config.baseUrl): Promise<string> => {
+    const { nodeClient } = await this.createNodeClientFromHeaders(cookie);
+    await nodeClient.signOut(redirectUri);
+
+    if (!this.navigateUrl) {
+      // Not expected to happen
+      throw new Error('navigateUrl is not set');
+    }
+
+    return this.navigateUrl;
+  };
+
+  handleSignInCallback = async (
+    cookie: string,
+    callbackUrl: string
+  ): Promise<string | undefined> => {
+    const { nodeClient, session } = await this.createNodeClientFromHeaders(cookie);
+
+    await nodeClient.handleSignInCallback(callbackUrl);
+    return session.getValues?.();
+  };
+
+  getLogtoContext = async (cookie: string, config: GetContextParameters = {}) => {
+    const { nodeClient } = await this.createNodeClientFromHeaders(cookie);
+    const context = await nodeClient.getContext(config);
+
+    return context;
+  };
+
+  private async createNodeClientFromHeaders(cookie?: string) {
+    const session = await createSession(
+      {
+        secret: this.config.cookieSecret,
+        crypto,
+      },
+      cookie ?? ''
+    );
```

Comment:
why cookie and setCookie are optional in this case? does server actions have some magic here?

Reply:
`cookie` is required, fixed. In server actions, it is unable to set cookies inside the SDK (can not set "use server" in library), so we use "getCookies" and then set cookies outside the sdk.